### PR TITLE
Updated test cases which should works when Klaytn uses dynamic gas price policy

### PR DESCRIPTION
**Pull request by aeharvlee**
## Proposed changes

Test cases should work after Klaytn applied Dynamic fee policy.
For this, I changed unit price of Klaytn which is not 25 ston.

**1. Updated legacy test cases**
We should confirm that legacy test works fine when Klaytn activate dynamic gas price policy.

For this, gas price should be fetched by using gasProvider.getGasPrice().
And also increase gas limit because some test cases on recent Klaytn node requires more gas.

**2. Updated recent test cases**
We need to use caver.transaction.${txType}.create to create transaction instance because now we need to fetch dynamic gas price and set with that value.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/caver-java/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/caver-java)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: core/src/test/java/com/klaytn/caver/legacy/feature/CustomGasTest.java [hunk 1]
```diff
@@ -67,32 +70,32 @@ public void testCustomGasPriceWithValueTransfer() throws Exception {
                 sender.getAddress(),
                 to.getAddress(),
                 value,
-                GAS_PRICE,
+                gasProvider.getGasPrice(),
                 GAS_LIMIT
         );
-        assertEquals(tx.getGasPrice(), GAS_PRICE);
+        assertEquals(tx.getGasPrice(), gasProvider.getGasPrice());
```
Comment (jimni1222) [inline core/src/test/java/com/klaytn/caver/legacy/feature/CustomGasTest.java:76]: Before `gasProvider.getGasPrice` might be different with after processing the tx `gasProvider.getGasPrice`.
How about making a variable to set `suggestedGas = gasProvider.getGasPrice()` and compare with that?

File: core/src/test/java/com/klaytn/caver/legacy/feature/CustomGasTest.java [hunk 2]
```diff
@@ -50,8 +52,9 @@ public class CustomGasTest {
 
     @BeforeClass
     public static void setup() throws Exception {
-        CHAIN_ID = ChainId.BAOBAB_TESTNET;
-        caver = Caver.build("https://api.baobab.klaytn.net:8651/");
+        caver = Caver.build("http://localhost:8551");
+        chainId = caver.rpc.klay.getChainID().send().getValue();
```
Comment (jimni1222) [inline core/src/test/java/com/klaytn/caver/legacy/feature/CustomGasTest.java:56]: Is this used?

Comment (aeharvlee) [reply core/src/test/java/com/klaytn/caver/legacy/feature/CustomGasTest.java:56]: Oops.. chainId is not used. I'll remove it.

File: core/src/test/java/com/klaytn/caver/legacy/feature/CustomGasTest.java [hunk 3]
```diff
@@ -35,11 +36,11 @@
 import java.math.BigInteger;
 
 public class CustomGasTest {
-    private static final BigInteger GAS_PRICE = BigInteger.valueOf(10L);
+    private static BigInteger gasPrice;
```
Comment (jimni1222) [inline core/src/test/java/com/klaytn/caver/legacy/feature/CustomGasTest.java:39]: Maybe this also not used?

Comment (aeharvlee) [reply core/src/test/java/com/klaytn/caver/legacy/feature/CustomGasTest.java]: Thanks. I checked other variables which is not used and remove them.

**Review by jimni1222 — APPROVED** [review]
LGTM!!!!

**Pull request merged**